src/internal/packager/helm/chart.go

```diff
@@ -118,6 +123,33 @@ func (h *Helm) InstallOrUpgradeChart(ctx context.Context) (types.ConnectStrings,
 		return nil, "", installErr
 	}
 
+	resourceList, err := h.actionConfig.KubeClient.Build(bytes.NewBufferString(release.Manifest), true)
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to build the resource list: %w", err)
+	}
+
+	healthChecks := []v1alpha1.NamespacedObjectKindReference{}
+	for _, resource := range resourceList {
+		apiVersion, kind := resource.Object.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+		healthChecks = append(healthChecks, v1alpha1.NamespacedObjectKindReference{
+			APIVersion: apiVersion,
+			Kind:       kind,
+			Name:       resource.Name,
+			Namespace:  resource.Namespace,
+		})
+	}
+	if !h.chart.NoWait {
+		// Ensure we don't go past the timeout by getting the time since the helm operation started
+		healthCheckTimeout := h.timeout - time.Since(helmOpStart)
```

Comment:
I understand the idea but it seems like it makes things complicated. Could we create a single context with timeout before install chart, that is used for both chart installing and health checking?

Reply:
I went back and forth on this. The retry loop makes context a bit complicated here. We want to initialize the timeout during each retry, but not cancel until the end of the function. I changed it the code to use the context, lmk what you think.